aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu

```diff
@@ -436,10 +436,9 @@ SparseTensor& add_out_sparse_cuda(const SparseTensor& t, const SparseTensor& src
 
   Tensor t_values_ = t._values().to(commonDtype);
   Tensor s_values_ = src._values().to(commonDtype);
-
-  AT_DISPATCH_ALL_TYPES_AND2(
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND2(
     at::ScalarType::Half, at::ScalarType::BFloat16, commonDtype, "add_out_sparse_cuda", [&] {
-      if (value.to<scalar_t>() != static_cast<scalar_t>(1)) {
+      if (value.to<scalar_t>() != scalar_t(1)) {
```

Comment:
We should continue using the `static_cast` here so please revert this change back

Reply:
I tried to maintain that,  but when `static_cast` is enabled it doesn't compile with complex data type.

Reply:
Just an idea, if `c10::complex` (I presume this corresponds to `scalar_t` if complex) would define a constructor using `int` as input, wouldn't `static_cast` work then?

Reply:
See, for instance, https://stackoverflow.com/questions/7312745/static-cast-for-user-defined-types

Reply:
Looks like this still needs to be addressed?

Reply:
I made the changes where it was possible, I was able to use `static_cast` with complex dtype when `cpp` compiler is used, however when `nvcc` compiler is used  this doesn't work, I tried to use the strategy suggested by pearu but it generate conflicts with other functions. So I guess the issue is with nvcc compiler as in other cases  😔